integrations/upgrade/index.test.ts

```diff
@@ -453,6 +453,130 @@ test(
   },
 )
 
+test(
+  'migrate imports with `layer(…)`',
+  {
+    fs: {
+      'package.json': json`
+        {
+          "dependencies": {
+            "tailwindcss": "workspace:^",
+            "@tailwindcss/upgrade": "workspace:^"
+          }
+        }
+      `,
+      'tailwind.config.js': js`module.exports = {}`,
+      'src/index.css': css`
+        @import './base.css';
+        @import './components.css';
+        @import './utilities.css';
+        @import './mix.css';
+
+        @tailwind base;
+        @tailwind components;
+        @tailwind utilities;
+      `,
+      'src/base.css': css`
+        html {
+          color: red;
+        }
+      `,
+      'src/components.css': css`
+        @layer components {
+          .foo {
+            color: red;
+          }
+        }
+      `,
+      'src/utilities.css': css`
+        @layer utilities {
+          .bar {
+            color: red;
+          }
+        }
+      `,
+      'src/mix.css': css`
+        html {
+          color: blue;
+        }
+
+        @layer components {
+          .foo-mix {
+            color: red;
+          }
+        }
+
+        @layer utilities {
+          .bar-mix {
+            color: red;
+          }
+        }
+      `,
+    },
+  },
+  async ({ fs, exec }) => {
+    await exec('npx @tailwindcss/upgrade')
+
+    expect(await fs.dumpFiles('./src/**/*.css')).toMatchInlineSnapshot(`
+      "
+      --- ./src/index.css ---
+      @import './base.css' layer(base);
+      @import './components.css';
+      @import './utilities.css';
+      @import './mix.css';
+
+      @import 'tailwindcss';
+
+      /*
+        The default border color has changed to \`currentColor\` in Tailwind CSS v4,
+        so we've added these compatibility styles to make sure everything still
+        looks the same as it did with Tailwind CSS v3.
+
+        If we ever want to remove these styles, we need to add an explicit border
+        color utility to any element that depends on these defaults.
+      */
+      @layer base {
+        *,
+        ::after,
+        ::before,
+        ::backdrop,
+        ::file-selector-button {
+          border-color: var(--color-gray-200, currentColor);
+        }
+      }
+
+      --- ./src/base.css ---
+      html {
+        color: red;
+      }
+
+      --- ./src/components.css ---
+      @utility foo {
+        color: red;
+      }
+
+      --- ./src/mix.css ---
+      @utility foo-mix {
+        color: red;
+      }
+
+      @utility bar-mix {
+        color: red;
+      }
+
+      html {
+        color: blue;
+      }
```

Comment:
Something to think about: should this file be split into an import _with_ the `layer(…)` and one with the utilities to ensure that this `html {}` base CSS is in the correct spot. 🤔

Reply:
I think this is right, otherwise `html {}` would be in the correct spot. I implemented it in the followup commits.